From 2e59d65c55a3814040715868020636cf28bd7b12 Mon Sep 17 00:00:00 2001
From: Riddle Hsu <riddlehsu@google.com>
Date: Tue, 26 Jun 2018 18:03:05 +0800
Subject: [PATCH 17/18] Get new surface if it was released from
 setWindowStopped

If there is no visibility change, when the stopped state
changed from true to false, there is no valid surface to use.

Bug: 80170221
Test: Sample app in bug link
Test: atest ViewRootImplTest
Change-Id: I2324ed32980aae5a3ec2efbd612e6c143c9a0022
Merged-In: I2324ed32980aae5a3ec2efbd612e6c143c9a0022
---
 core/java/android/view/ViewRootImpl.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/java/android/view/ViewRootImpl.java b/core/java/android/view/ViewRootImpl.java
index edb525a92feb..be96aafb3c23 100644
--- a/core/java/android/view/ViewRootImpl.java
+++ b/core/java/android/view/ViewRootImpl.java
@@ -1360,6 +1360,7 @@ public final class ViewRootImpl implements ViewParent,
                 renderer.setStopped(mStopped);
             }
             if (!mStopped) {
+                mNewSurfaceNeeded = true;
                 scheduleTraversals();
             } else {
                 if (renderer != null) {
-- 
2.17.1

